packages/core/src/__tests__/ceramic-pinning.test.ts

```diff
@@ -141,12 +141,11 @@ describe('Ceramic stream pinning', () => {
       anchor: false,
       publish: false,
     })
-    await expect(isPinned(ceramic, stream.id)).resolves.toBeFalsy()
+    expect(isPinned(ceramic, stream.id)).resolves.toBeFalsy()
```

Comment:
why no `await` here? @stephhuynh18

Reply:
I am reaching an odd bug where the tests will hang indefinitely on this test. Specifically the `isPinned()` on line 148. I removed the `awaits` and the tests behaved accordingly (ex. if i change `toBeFalsy()` to `toBeTruthy` the test fails, no changes means the tests pass). But looking at the documentation I am wrong. But why does it hang on `isPinned` on only this line but not on the other lines.... If you have any insights pls let me know

Reply:
@ukstv So i did some investigating and this is what I found:
`stream.update()`  --> `ceramic.applyCommit()` --> `repo.applyCommit()` 
It then calls
`stateManager.applyCommit()` --> `repo.load()` --> `repo.handlePinOpts()` --> `repo.unpin()`
After the above completes the following gets called 
`repo.applyWriteOpts()` --> `repo.handlePinOpts()` --> `repo.unpin()`

So essentially `repo.unpin()` gets called twice. This functions is async and so this double call may be whats confusing jest causing it to hang even after the test successfully completes . If I comment out one of `repo.handlePinOpts` calls the test passes and doesn't hang. If i don't unpin in the test, the test passes and doesn't hang.

I'm not sure if I am right so please help me! The duplicated call to unpin is unexpected

Reply:
well i guess nvm the tests passed....

Reply:
@stephhuynh18 Your finding is true, so https://github.com/ceramicnetwork/js-ceramic/pull/1872